compose/src/main/kotlin/io/github/kakaocup/compose/node/element/lazy_list/KLazyListNode.kt

```diff
@@ -0,0 +1,83 @@
+package io.github.kakaocup.compose.node.element.lazy_list
+
+import androidx.compose.ui.test.ExperimentalTestApi
+import androidx.compose.ui.test.SemanticsMatcher
+import androidx.compose.ui.test.SemanticsNodeInteractionsProvider
+import androidx.compose.ui.test.filterToOne
+import androidx.compose.ui.test.onChildren
+import io.github.kakaocup.compose.node.builder.NodeMatcher
+import io.github.kakaocup.compose.node.builder.ViewBuilder
+import io.github.kakaocup.compose.node.core.BaseNode
+
+/**
+ * Node class with special api to test Lazy List (LazyColumn or LazyRow)
+ */
+class KLazyListNode(
+    semanticsProvider: SemanticsNodeInteractionsProvider,
+    nodeMatcher: NodeMatcher,
+    itemTypeBuilder: KLazyListItemBuilder.() -> Unit,
+    val positionMatcher: (position: Int) -> SemanticsMatcher
+) : BaseNode<KLazyListNode>(semanticsProvider, nodeMatcher) {
+    val semanticsMatcher = nodeMatcher.matcher
+    val itemTypes = KLazyListItemBuilder().apply(itemTypeBuilder).itemTypes
+
+    /**
+     * Constructs node class with node interaction from given ViewBuilder
+     *
+     * @param semanticsProvider Provides main entry point into testing
+     * @param viewBuilderAction ViewBuilder which will result in view's interaction
+     * @param itemTypeBuilder Lambda with receiver where you pass your item providers
+     * @param positionMatcher Lambda which finds node by given position
+     *
+     * @see ViewBuilder
+     */
+    constructor(
+        semanticsProvider: SemanticsNodeInteractionsProvider,
+        viewBuilderAction: ViewBuilder.() -> Unit,
+        itemTypeBuilder: KLazyListItemBuilder.() -> Unit,
+        positionMatcher: (position: Int) -> SemanticsMatcher
+    ) : this(
+        semanticsProvider = semanticsProvider,
+        nodeMatcher = ViewBuilder().apply(viewBuilderAction).build(),
+        itemTypeBuilder = itemTypeBuilder,
+        positionMatcher = positionMatcher
+    )
+
+    /**
+     * Performs given actions/assertion on child at given position
+     *
+     * @param T Type of item at given position. Must be registered via constructor.
+     * @param position Position of item in lazy list
+     * @param function Tail lambda which receiver will be matched item with given type T
+     */
+    @ExperimentalTestApi
+    inline fun <reified T : KLazyListItemNode> childAt(
+        position: Int,
+        function: T.() -> Unit
+    ) {
+        val provideItem = itemTypes.getOrElse(T::class) {
+            throw IllegalStateException("${T::class.java.simpleName} did not register to KLazyListNode")
+        }.provideItem
+
+        runCatching { performScrollToIndex(position) }
```

Comment:
You skip the possible exception here. What to do if something went wrong during the scrolling? How to provide enough info to a user?

Reply:
This is just an example for the user how they can match a semantics node by id in a more convenient way. This could be done by simply using the TestTag as a string e.g. "item 1"